src/Components/Assets/AssetFilter.tsx

```diff
@@ -40,7 +73,39 @@ function AssetFilter(props: any) {
     pathParams: { id: facilityId },
     onResponse: ({ res, data }) => {
       if (res?.status === 200 && data) {
-        setFacility(data);
+        const transformedData = {
+          id: Number(data.id),
+          name: data.name,
+          read_cover_image_url: data.read_cover_image_url,
+          facility_type: String(data.facility_type),
+          address: data.address,
+          features: data.features,
+          location: {
+            latitude: Number(data.latitude),
+            longitude: Number(data.longitude),
+          },
+          oxygen_capacity: data.oxygen_capacity,
+          phone_number: data.phone_number,
+          type_b_cylinders: data.type_b_cylinders,
+          type_c_cylinders: data.type_c_cylinders,
+          type_d_cylinders: data.type_d_cylinders,
+          middleware_address: data.middleware_address,
+          expected_type_b_cylinders: data.expected_type_b_cylinders,
+          expected_type_c_cylinders: data.expected_type_c_cylinders,
+          expected_type_d_cylinders: data.expected_type_d_cylinders,
+          expected_oxygen_requirement: data.expected_oxygen_requirement,
+          local_body_object: data.local_body_object,
+          district_object: data.district_object,
+          state_object: data.state_object,
+          ward_object: data.ward_object,
+          modified_date: data.modified_date,
+          created_date: data.created_date,
+          state: data.state,
+          district: data.district,
+          local_body: data.local_body,
+          ward: data.ward,
+        };
+        setFacility(transformedData);
```

Comment:
What exactly are you transforming? Types? Avoid casting to wrong types.

```suggestion
        setFacility(data);
```